# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cache/CacheStorage.cpp
# Commit: 0e68db4ad6af
# Full Hash: 0e68db4ad6afe1f43d97533316d80e5aadd96082
# Author: Luca Greco <lgreco@mozilla.com>
# Date: 2020-07-01 21:32:12
# Regressor Bug: 1609920
# File Overlap Count: 2
# Description:
#   Bug 1609920 - part 1: Allow the WebExtension Framework to register a moz-extension service worker. r=dom-workers-and-storage-reviewers,asuth,mixedpuppy
#   
#   - Adds the new about:config pref "extensions.backgroundServiceWorker.enabled" (currently defaults to false).
#   - Adds the background.service_worker property to the manifest JSON schema definition
#   - Locks background.service_worker manifest property behind the new preference
# ==============================================================================

diff -r 2a107eabe00c -r 0e68db4ad6af dom/cache/CacheStorage.cpp
--- a/dom/cache/CacheStorage.cpp	Mon Jun 29 16:18:44 2020 +0000
+++ b/dom/cache/CacheStorage.cpp	Wed Jun 10 12:13:29 2020 +0000
@@ -27,6 +27,7 @@
 #include "mozilla/ipc/PBackgroundChild.h"
 #include "mozilla/ipc/PBackgroundSharedTypes.h"
 #include "mozilla/StaticPrefs_dom.h"
+#include "mozilla/StaticPrefs_extensions.h"
 #include "nsContentUtils.h"
 #include "mozilla/dom/Document.h"
 #include "nsIGlobalObject.h"
@@ -118,6 +119,11 @@
     return true;
   }
 
+  if (StaticPrefs::extensions_backgroundServiceWorker_enabled_AtStartup() &&
+      scheme.LowerCaseEqualsLiteral("moz-extension")) {
+    return true;
+  }
+
   uint32_t hostPos;
   int32_t hostLen;
 